catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileAlterationObserver.java

```diff
@@ -37,41 +37,84 @@
  * directory being monitored, and call the {@link AsyncFileAlterationListener}'s corresponding
  * methods
  *
- * @apiNote The AsyncFileAlterationListener is expected to either call the callback function passed
- *     with {@code null} or pass the callback function into an {@link org.apache.camel.Exchange}.
- * @apiNote If this is loaded from a file after the system has been shut down, the user
- *     <strong>MUST</strong> call {@code clearCache()} in order to clear the currently processing
- *     request.
+ * @see AsyncFileAlterationListener
  */
 public class AsyncFileAlterationObserver implements Serializable {
 
-  private final AsyncFileEntry rootFile;
-  private AsyncFileAlterationListener listener = null;
-  private final Set<AsyncFileEntry> processing = new ConcurrentSkipListSet<>();
+  @Nullable private final AsyncFileEntry rootFile; //  In case GSON returns it.
+  @Nullable private transient AsyncFileAlterationListener listener = null;
+  private final transient Set<AsyncFileEntry> processing = new ConcurrentSkipListSet<>();
   private static final Logger LOGGER = LoggerFactory.getLogger(AsyncFileAlterationObserver.class);
+  private final transient Object listenerLock = new Object();;
 
   public AsyncFileAlterationObserver(File fileToObserve) {
+    if (fileToObserve == null) {
+      throw new IllegalArgumentException();
+    }
     rootFile = new AsyncFileEntry(fileToObserve);
   }
 
-  //  Only for testing
+  //  For GSON serialization;
+  private AsyncFileAlterationObserver() {
+    rootFile = null;
+  }
+
+  public void onLoad() {
+    if (rootFile != null) rootFile.initialize();
+  }
+
+  //  Returns true on errors;
+  private boolean initHelper(AsyncFileEntry entry) {
```

Comment:
✏️ I think returning true when there are errors is a little counter-intuitive since usually methods return true if successful. Like you did with the method below.

Reply:
👍 I'll change the boolean logic to reflect that